lego/apps/users/views/memberships.py

```diff
@@ -32,4 +34,12 @@ def get_queryset(self):
 
     def create(self, request, *args, **kwargs):
         request.data["abakus_group"] = kwargs["group_pk"]
+        user_groups = [
+            membership.abakus_group.name for membership in request.user.memberships
+        ]
+        group_id = kwargs["group_pk"]
+        group = AbakusGroup.objects.get(pk=group_id)
+        if group.type == constants.GROUP_INTEREST and "Abakus" not in user_groups:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
```

Comment:
This could be simplified slightly by using the `is_abakus_member` property:) (as defined in https://github.com/webkom/lego/blob/master/lego/apps/users/models.py#L213-L215)

```suggestion
        group = AbakusGroup.objects.get(pk=kwargs["group_pk"])
        if group.type == constants.GROUP_INTEREST and not request.user.is_abakus_member:
            return Response(status=status.HTTP_403_FORBIDDEN)

```

Reply:
I tried using the is_abakus_member property as suggested but non-students were still able to join. Switching to has_grade_group resolved the issue, though I'm unsure if that was the intended approach.

Reply:
Ahh I see, I suggested you use `is_abakus_member` because that was what your code checked previously. However, a user can be an abakus member without being a student, so if what you wanted to accomplish was to only allow students to register to groups then you're right, `has_grade_group` would be a better approach!